src/core/dom/static-template.js

```diff
@@ -81,7 +81,27 @@ function html(strings) {
  */
 function createNode(container, strings) {
   devAssert(strings.length === 1, 'Improper html template tag usage.');
-  container./*OK*/ innerHTML = strings[0];
+  devAssert(
+    Array.isArray(strings) || hasOwn(strings, 'raw'),
+    'Invalid template strings array'
+  );
+
+  if (self.trustedTypes && self.trustedTypes.createPolicy) {
+    (function () {
```

Comment:
could you explain to me why there is a need for an IIFE here?

Reply:
We thought limiting the scope with IIFE would make sure that the Trusted Types policy doesn't leak outside the intended scope, so it can't be used to bless arbitrary strings as `TrustedHTML`s anywhere else. Trusted Types would lose its security guarantees if any arbitrary value can be converted to a `TrustedHTML`, since this would make calls Trusted Types compatible when they could potentially be unsafe. 

Admittedly, this was an extra cautious step since I'm not fully familiar with the AMP codebase and I wasn't sure if the TT policy could leak elsewhere. Happy to change the implementation to not use IIFE if there's no risk of leaking the policy.

Reply:
gotcha it was in the issue description "generation happens in an anonymous function to minimize the risk of the Trusted Types policy leaking and being used elsewhere to generate TrustedHTML".

would you mind adding this in as a comment. thanks!

Reply:
gotcha. thanks again for explaining, in this scenario since you use `const` it should be block scoped to the `if` statement so we need the IIFE. if you could remove it ill go ahead and merge this right after. apologies for the back and forth